lib/p2p/Pool.ts

```diff
@@ -65,16 +69,21 @@ class Pool extends EventEmitter {
     this.connected = false;
   }
 
-  public addOutbound = async (address: string, port: number): Promise<Peer> => {
-    const socketAddress = new SocketAddress(address, port);
+  /**
+   * Attempt to add an outbound peer by connecting to a given SocketAddress.
+   * Throws an error if a connection already exists for the provided address.
+   * If the connection is successful, send a [[GetHostsPacket]] to discover new XUD hosts to connect to.
+   */
+  public addOutbound = async (socketAddress: SocketAddress): Promise<Peer> => {
     if (this.peers.has(socketAddress)) {
-      const err = errors.ADDRESS_ALREADY_CONNECTED(address.toString());
+      const err = errors.ADDRESS_ALREADY_CONNECTED(socketAddress.address);
       this.logger.info(err.message);
       throw err;
     }
 
     const peer = Peer.fromOutbound(socketAddress);
     await this.tryConnectPeer(peer);
+    peer.sendPacket(new GetHostsPacket(null));
```

Comment:
Our current design hashes the packet by the body. It means that the body should be unique in some way. that's why I added `ts` field to ping/pong packets body - not a perfect solution, but better then leaving it empty.

Reply:
Yeah I considered this but wasn't sure what to go with. There's a similar issue with the get orders packet and this will be the case with any p2p request that doesn't have anything meaningful to put in the body. I figure for pings the time is useful to determine latency.

Maybe there's some refactoring that can be done here, treating the empty request packets (get orders, get hosts) similarly to the way we treat the Hello packet currently. If that sounds reasonable to you, I'll see what I can come up with.

Reply:
I think this would be related to #208 as well.

Reply:
I don’t think it’s related to how we handle `Hello` packet, because here we have a typical request/response scenarios. 

I think that a timestamp field should be sufficient, because the hash need to be unique only from the sender perspective (he's the one who's waiting for the response). This could be done on the packet type class, so when instantiating a new packet instance, you don't need to provide anything, but the constructor add it.

Reply:
Other option: adding a uuid.
I used timestamp in case it will be useful for other things as well.